tests/test_grid.py

```diff
@@ -55,22 +58,40 @@ def test_make_square_grid():
     # check if correct number of points were created
     assert len(test.keys()) == 100
 
-    # check that distance between neighboring points is euqal for all integers
-    # (this works but is a bit long)
-    for x in range(9):
-        for y in range(9):
-            distance = euclidean(
-                test[f"{x}-{y}"]["poly"][0], test[f"{x}-{y}"]["poly"][2]
-            )
-            distance2 = euclidean(
-                test[f"{x}-{y}"]["poly"][1], test[f"{x}-{y}"]["poly"][3]
-            )
-            assert distance == pytest.approx(distance2)
-    # TODO check that angle between points = 90 deg
-
-
-def test_make_hex_grid():
-    """Test _make_hex_grid()."""
+    # check distance between corner points is the same
+    distance = euclidean(test[f"{x}-{y}"]["poly"][0], test[f"{x}-{y}"]["poly"][2])
+    distance2 = euclidean(test[f"{x}-{y}"]["poly"][1], test[f"{x}-{y}"]["poly"][3])
+    assert distance == pytest.approx(distance2)
+
+    # check angle between corner points is the same
+    from math import atan2, degrees, pi
```

Comment:
A simpler approach would be to calculate the inner product of the two segments and check if it is zero. So, if AB and AC are numpy arrays that go from point A to B and point A to C respectively, then they will form a ±90 deg angle if:

```python
assert np.inner(AB, AC) == pytest.approx(0)
```
If you need specifically the angle (which you do for the hexagonal one), you can follow an approach like this: https://www.atqed.com/numpy-vector-angle

Reply:
I replaced the initial checks for angles with this approach, thanks for the suggestion